RecoLocalCalo/HcalRecAlgos/interface/HcalSiPMParametersGPU.h

```diff
@@ -0,0 +1,41 @@
+#ifndef RecoLocalCalo_HcalRecAlgos_interface_HcalSiPMParametersGPU_h
+#define RecoLocalCalo_HcalRecAlgos_interface_HcalSiPMParametersGPU_h
+
+#include "CondFormats/HcalObjects/interface/HcalSiPMParameters.h"
+
+#ifndef __CUDACC__
+#include "HeterogeneousCore/CUDAUtilities/interface/HostAllocator.h"
+#include "HeterogeneousCore/CUDACore/interface/ESProduct.h"
+#endif
+
+class HcalSiPMParametersGPU {
+public:
+  struct Product {
+    ~Product();
+    int *type, *auxi1;
+    float *fcByPE, *darkCurrent, *auxi2;
+  };
+
+#ifndef __CUDACC__
+  // rearrange reco params
+  HcalSiPMParametersGPU(HcalSiPMParameters const &);
+
+  // will trigger deallocation of Product thru ~Product
+  ~HcalSiPMParametersGPU() = default;
+
+  // get device pointers
+  Product const &getProduct(cudaStream_t) const;
+
+private:
+  uint64_t totalChannels_;
+  std::vector<int, cms::cuda::HostAllocator<int>> type_, auxi1_;
+  std::vector<float, cms::cuda::HostAllocator<float>> fcByPE_, darkCurrent_, auxi2_;
+  /*
```

Comment:
If this line is not needed, it can be removed. Otherwise just add another comment line which explains why do you want to keep here even if  commented out